### PR TITLE
Update RegisterMandatoryFitPredicate to avoid double register.

### LINKED ISSUE #50360: Move node condition filter in scheduler.factory into a new predicates
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind cleanup
/sig scheduling

**Descriptions**:

According to the discussion at https://github.com/kubernetes/kubernetes/pull/50185 , we'll move the node condition filter into a new predicates; and use feature gate to turn-off/on it.

/cc @gmarek @kubernetes/sig-scheduling-misc

### PR DESCRIPTION
**Pull request by k82cn**
**What this PR does / why we need it**:
In https://github.com/kubernetes/kubernetes/pull/50362 , we introduced `RegisterMandatoryFitPredicate` to make some predicates always included by scheduler. This PRs is to improve it by avoiding double register: `RegisterFitPredicate` and `RegisterMandatoryFitPredicate` 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #50360 

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by zhangxiaoyu-zidif** [comment]
The title has a typo :)
s/doubel/double

**Review by zhangxiaoyu-zidif — CHANGES_REQUESTED** [review]
The title has a typo :)
s/doubel/double

**Comment by k82cn** [comment]
@bsalamat , ping for review :).

File: plugin/pkg/scheduler/factory/plugins.go [hunk 1]
```diff
@@ -321,10 +322,12 @@ func getFitPredicateFunctions(names sets.String, args PluginFactoryArgs) (map[st
 		predicates[name] = factory(args)
 	}
 
-	// Always include required fit predicates.
-	for name, factory := range mandatoryFitPredicateMap {
-		if _, found := predicates[name]; !found {
-			predicates[name] = factory(args)
+	// Always include mandatory fit predicates.
+	for name, mandatory := range mandatoryFitPredicates {
+		if mandatory {
```
Comment (bsalamat) [inline plugin/pkg/scheduler/factory/plugins.go]: The current logic is correct, but I would remove checking the value of `mandatory`. `mandatoryFitPredicate` is basically a set. Anything in that set can be considered a mandatory predicate, regardless of the value of `bool`.

Comment (k82cn) [inline plugin/pkg/scheduler/factory/plugins.go]: Good point ! Will update it today :).

Comment (k82cn) [inline plugin/pkg/scheduler/factory/plugins.go]: done :).

File: plugin/pkg/scheduler/factory/plugins.go [hunk 2]
```diff
@@ -323,11 +323,9 @@ func getFitPredicateFunctions(names sets.String, args PluginFactoryArgs) (map[st
 	}
 
 	// Always include mandatory fit predicates.
-	for name, mandatory := range mandatoryFitPredicates {
-		if mandatory {
-			if factory, found := fitPredicateMap[name]; found {
-				predicates[name] = factory(args)
-			}
+	for _, name := range mandatoryFitPredicates.List() {
```
Comment (bsalamat) [inline plugin/pkg/scheduler/factory/plugins.go]: Please do not use `List()` here. It first copies all the items of the map (set) into a slice and then sorts the items and returns the slice. We don't need a sorted list. Instead, just use:
`for name, _ := range mandatoryFitPredicates {`

Comment (k82cn) [inline plugin/pkg/scheduler/factory/plugins.go]: ops :). and done .

**Comment by bsalamat** [comment]
/lgtm

Thanks, @k82cn!

**Pull request merged**